### PR TITLE
zq/fix_declaration_for_some_func

### PR DESCRIPTION
**Pull request by NeosZhang**
## Motivation and Context

修复optimizer类相关函数的描述以及对应的python函数的返回值

## Description

optimizer类的函数(sgd, adam, adamw, adadelta, rmsprop)中的param_grad不会被修改，应该是const的
修复optimizer类相关函数的描述以及对应的python函数的返回值

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: proto/include/diopi/functions.h [hunk 1]
```diff
@@ -2405,7 +2405,7 @@ DIOPI_API diopiError_t diopiReciprocalInp(diopiContextHandle_t ctx, diopiTensorH
  * @param[in] step step. type = [int64].
  * @param[in] amsgrad whether to use the AMSGrad variant of this algorithm from the paper `On the Convergence of Adam and Beyond`_.
  */
-DIOPI_API diopiError_t diopiAdamW(diopiContextHandle_t ctx, diopiTensorHandle_t input, diopiTensorHandle_t grad, diopiTensorHandle_t exp_avg,
```
Comment (yangbofun) [inline proto/include/diopi/functions.h:2408]: 这个input改下名字，叫params， 把文档中对input说明的in，改为inout。

Comment (NeosZhang) [reply proto/include/diopi/functions.h:2408]: 嗯嗯 好的

Comment (NeosZhang) [reply proto/include/diopi/functions.h:2408]: done，同时修复了其他optimizer类(sgd, adam, adamw, adadelta, rmsprop)函数的声明和注释

File: impl/camb/device_configs.py [hunk 2]
```diff
@@ -1118,7 +1118,7 @@
                 {
                     # can't get correct result
                     "ins": ['param', 'param_grad'],
-                    "dtype": [Skip(np.float16)],
+                    "dtype": [Skip(np.float64)],
```
Comment (NeosZhang) [inline impl/camb/device_configs.py:1121]: 查看寒武纪CNNL文档，cnnlApplyAdadelta只支持float16和float32
[参考链接](https://www.cambricon.com/docs/sdk_1.15.0/cambricon_cnnl_1.21.1/developer_guide/cnnl_api/api/applyadadelta.html?highlight=cnnlapplyadadelta#_CPPv417cnnlApplyAdadelta12cnnlHandle_tK22cnnlTensorDescriptor_tPvK22cnnlTensorDescriptor_tPvK22cnnlTensorDescriptor_tPvK22cnnlTensorDescriptor_tPKvPKvPKvPKv)
image

File: scripts/filter_ci.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ def get_run_result(pr_number):
     response = requests.get(api_url, headers=headers)
     if response.status_code == 200:
         pr_files = response.json()
-        if "diopi_configs.py" in str(pr_files):
+        if "diopi_configs.py" in str(pr_files) or "customized_test.py" in str(pr_files):
```
Comment (NeosZhang) [inline scripts/filter_ci.py:24]: 修改CI逻辑，当diopi_configs.py或customized_test.py被修改时都将触发gen_data

**Review by yangbofun — APPROVED** [review]

**Pull request merged**